Modules/_functoolsmodule.c

```diff
@@ -322,6 +366,7 @@ partial_setstate(partialobject *pto, PyObject *state)
         Py_INCREF(dict);
 
     Py_SETREF(pto->fn, fn);
+    pto->use_fastcall = _PyObject_HasFastCall(fn);
```

Comment:
Does it before `Py_SETREF()`. The partial object should be consistent at the time of destroying old callable.